### PR TITLE
Adding R package Rvcg

### PR DESCRIPTION
**Pull request by dipterix**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by dipterix** [comment]
Ping @conda-forge/r : This PR is for R package `Rvcg`, ready for review.

I confirm I want to be the maintainer of this recipe.

**Comment by dipterix** [comment]
@conda-forge/help-r

**Comment by dpryan79** [comment]
@conda-forge/staged-recipes This is ready for review

File: recipes/r-rvcg/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    - {{ posix }}coreutils         # [win]
+    - {{ posix }}zip               # [win]
+    - cross-r-base {{ r_base }}    # [build_platform != target_platform]
+
+  host:
+    - r-base
+    - r-rcpp
+    - r-rcpparmadillo
+    - r-rcppeigen
+    - _openmp_mutex                 # [linux]
```
Comment (mfansler) [inline recipes/r-rvcg/meta.yaml:49]: Why this dependency? More typical is:

```suggestion
    - libgomp                       # [linux]
```

File: recipes/r-rvcg/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  number: 0
+  # no skip
+
+  # This is required to make R link correctly on Linux.
+  rpaths:
+    - lib/R/lib/
+    - lib/
+  
+  missing_dso_whitelist:
+    - '*/R.dll'
```
Comment (mfansler) [inline recipes/r-rvcg/meta.yaml:28]: I've never seen this before. Not saying it needs to change, but is there a precedent?

**Review by mfansler — CHANGES_REQUESTED** [review]
I've suggested some improvements.

File: recipes/r-rvcg/LICENSE [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Please see the included file COPYRIGHTS for per-file (or per-directory)
```
[the remark this answers is not in the document]
Comment (dipterix) [reply recipes/r-rvcg/LICENSE:1]: Will do

File: recipes/r-rvcg/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    - r-rcpp
+    - r-rcpparmadillo
+    - r-rcppeigen
+    - _openmp_mutex                 # [linux]
+    - llvm-openmp                   # [osx]
+
+  run:
+    - r-base
+    - {{ native }}gcc-libs         # [win]
+    - m2w64-gcc-libs-core          # [win]
```
[the remark this answers is not in the document]
Comment (dipterix) [reply recipes/r-rvcg/meta.yaml:55]: The package raises error when checking if `m2w64-gcc-libs-core` is missing

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=660876&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=2fb5b0a7-737c-5f5c-8f3f-f6db6174bacf&l=3632

```
  ERROR (r-rvcg,lib/R/library/Rvcg/libs/x64/Rvcg.dll): Needed DSO Library/mingw-w64/bin/libgcc_s_seh-1.dll found in ['m2w64-gcc-libs-core']
  ERROR (r-rvcg,lib/R/library/Rvcg/libs/x64/Rvcg.dll): .. but ['m2w64-gcc-libs-core'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

File: recipes/r-rvcg/meta.yaml (hunk 2 above)
Comment (dipterix) [reply recipes/r-rvcg/meta.yaml:28]: Please check here:

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=660876&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=2fb5b0a7-737c-5f5c-8f3f-f6db6174bacf&l=3629

File: recipes/r-rvcg/meta.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (mfansler) [reply recipes/r-rvcg/meta.yaml:55]: That was before the whitelisting of `R.dll`, so I would still like to see it tried. If it still fails, then at least it should instead be:

```suggestion
    - {{ native }}gcc-libs-core    # [win]
```

[the remark this answers is not in the document]
Comment (dipterix) [reply recipes/r-rvcg/meta.yaml:55]: Gotcha!

I'm trying to sort out my TODO list:

0. replace `_openmp_mutex` with `libgomp`
1. whitelisting R.dll, remove `m2w64-gcc-libs-core`
2. If 1 fails, try `{{ native }}gcc-libs-core`

Is this what I'm supposed to do?

[the remark this answers is not in the document]
Comment (mfansler) [reply recipes/r-rvcg/meta.yaml:55]: Yes. That sounds good - plus the cleaning up of the licenses (remove the extra file; add COPYRIGHTS to list). I won't protest the `R.dll` whitelist, but I'm also not a merger on `staged-recipes`, just part of the **help-r** team.

**Comment by zarquon42b** [comment]
Hey guys, pretty cool that @dipterix is porting my R-package to conda. He told me you will need me to state here that I would like to be added as maintainer of that package.

**Comment by dipterix** [comment]
@mfansler @conda-forge/help-r 

I have edited the recipe (commit https://github.com/conda-forge/staged-recipes/pull/22073/commits/cd6b4caf96df4d8cb32edfc81b41c4823771b227) as required.

In addition, I added the original R package maintainer @zarquon42b to the maintainer list. Please see his comment above.

I believe to my best knowledge this package is ready to review. 

Thanks : )

**Review by mfansler — APPROVED** [review]
@conda-forge/staged-recipes this looks ready to merge

**Comment by dipterix** [comment]
ping https://github.com/orgs/conda-forge/teams/staged-recipes

File: recipes/r-rvcg/meta.yaml (hunk 2 above)
Comment (dipterix) [reply recipes/r-rvcg/meta.yaml:28]: Ah great, that makes sense. I tried compiling on my windows machine and Rvcg loads normally with R.dll whitelisted. I guess this is fine? (If so, can I mark this conversation as resolved?

Comment (mfansler) [reply recipes/r-rvcg/meta.yaml:28]: Yes, but I think I marked it already. ✅ And thanks for testing locally!

**Comment by mfansler** [comment]
Why is whitelist being removed? The precedent I pointed out is that someone in Conda Forge core is recommending that all recipes should whitelist this when generating CRAN recipes.

**Comment by dipterix** [comment]
> Why is whitelist being removed? The precedent I pointed out is that someone in Conda Forge core is recommending that all recipes should whitelist this when generating CRAN recipes.

Sorry for the confusion. I misunderstood your comment "Yes, but I think I marked it already." posted 4 hours ago. I thought "it" was whitelisting (marking R.dll in whitelisting already) and forgot the context (of which "it" meant the remarks of the PR suggestions). I have add R.dll back just now.

Again sorry, no bad intention here.

**Comment by dipterix** [comment]
@conda-forge/staged-recipes , @conda-forge/help-r ready for review & merge : )

**Comment by dipterix** [comment]
ping

**Pull request merged**